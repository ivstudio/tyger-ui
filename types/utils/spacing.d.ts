export declare const spacing: {
    readonly '0': "0px";
    readonly '2': "2px";
    readonly '4': "4px";
    readonly '8': "8px";
    readonly '12': "12px";
    readonly '16': "16px";
    readonly '24': "24px";
    readonly '32': "32px";
    readonly '40': "40px";
    readonly '48': "48px";
    readonly '64': "64px";
    readonly '80': "80px";
    readonly '96': "96px";
    readonly '160': "160px";
};
export declare type TSpacing = keyof typeof spacing;
