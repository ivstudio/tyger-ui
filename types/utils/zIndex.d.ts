export declare const zIndex: {
    readonly mobileStepper: 1000;
    readonly appBar: 1100;
    readonly drawer: 1200;
    readonly modal: 1300;
    readonly snackbar: 1400;
    readonly tooltip: 1500;
};
export declare type TZIndex = keyof typeof zIndex;
