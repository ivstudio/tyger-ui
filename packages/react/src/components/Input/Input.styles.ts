import styled from 'styled-components';

import { textFieldBase, textFieldSpacing } from '../TextField/TextField.styles';
import { InputComponentProps } from './Input.types';

export const textFieldHeight = {
    sm: 32,
    md: 40,
    lg: 48,
};

export const InputComponent = styled.input.attrs({
    novalidate: true,
})<InputComponentProps>`
    /* Base styles for all text fields */
    ${textFieldBase};

    /* Fix for number inputs in Firefox */
    &[type='number'] {
        -moz-appearance: textfield;
    }

    /* Set spacing based on input size */
    padding: ${({ sizeInput, startAdornment, endAdornment }) => {
        // if (startAdornment) {
        //     return `0 ${textFieldSpacing[sizeInput]} 0 ${textFieldSpacing[sizeInput]}`;
        // }

        // if (endAdornment) {
        //     return `0 ${textFieldSpacing[sizeInput]} 0 ${textFieldSpacing[sizeInput]}`;
        // }

        return textFieldSpacing[sizeInput];
    }};
    height: ${({ sizeInput }) => textFieldHeight[sizeInput]};
`;

// Define the fieldset
export const Fieldset = styled.fieldset`
    border: none;
    padding: 0;
    margin: 0;
`;

export const InputContainerComponent = styled.div`
    position: relative;
    height: 100%;
`;

export const Adornment = styled.div<{ start?: boolean; end?: boolean }>`
    position: absolute;
    left: 0;
    display: flex;
    align-items: center;
    height: 100%;

    ${({ start }) =>
        start && {
            left: 0,
        }};

    ${({ end }) =>
        end && {
            right: 0,
        }};
`;
