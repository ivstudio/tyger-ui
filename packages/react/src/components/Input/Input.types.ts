export type InputSizes = 'sm' | 'md' | 'lg';

export interface InputBase {
    error?: boolean;
    className?: string;
    size?: InputSizes;
    readOnly?: boolean;
    disabled?: boolean;
    label?: string;
    required?: boolean;
}

export interface InputProps
    extends InputBase,
        Omit<React.ComponentProps<'input'>, 'size'> {
    size?: InputSizes;
    helperText?: string;
}

export interface InputComponentProps extends InputBase {
    sizeInput: InputSizes;
}
