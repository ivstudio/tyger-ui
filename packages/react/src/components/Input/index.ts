'use client';
export { default } from './Input';
