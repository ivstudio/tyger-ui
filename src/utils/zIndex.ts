export const zindex = {
    mobileStepper: 1000,
    appBar: 1100,
    drawer: 1200,
    modal: 1300,
    snackbar: 1400,
    tooltip: 1500,
};
export type TZIndex = keyof typeof zindex;
