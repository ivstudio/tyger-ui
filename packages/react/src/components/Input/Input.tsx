'use client';

import { forwardRef } from 'react';

import { Fieldset, HelperText, InputComponent, Label } from './Input.styles';
import { InputProps } from './Input.types';

const Input = forwardRef<HTMLInputElement, InputProps>(
    (props, forwardedRef) => {
        const {
            size = 'md',
            id,
            error,
            helperText,
            label,
            required = false,
            ...baseProps
        } = props;

        return (
            <Fieldset tag="fieldset" border={false}>
                {label && (
                    <Label htmlFor={id} required={required}>
                        {label}
                    </Label>
                )}
                <InputComponent
                    {...baseProps}
                    ref={forwardedRef}
                    id={id}
                    error={error}
                    aria-required={required}
                    aria-invalid={error}
                    sizeInput={size}
                />
                {error && (
                    <HelperText mt="4" mb="0">
                        {helperText}
                    </HelperText>
                )}
            </Fieldset>
        );
    }
);

Input.displayName = 'Input';
export default Input;
