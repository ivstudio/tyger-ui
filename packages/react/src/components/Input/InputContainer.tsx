'use client';

import { Adornment, InputContainerComponent } from './Input.styles';
import { ContainerProps } from './Input.types';

const InputContainer = ({
    hasAdornment,
    children,
    startAdornment = undefined,
    endAdornment = undefined,
}: ContainerProps) => {
    if (!hasAdornment) return <>{children}</>;
    return (
        <InputContainerComponent>
            {startAdornment && <Adornment start>{startAdornment}</Adornment>}
            <>{children}</>
            {endAdornment && <Adornment end>{endAdornment}</Adornment>}
        </InputContainerComponent>
    );
};

export default InputContainer;
