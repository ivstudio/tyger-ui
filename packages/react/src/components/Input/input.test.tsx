import React from 'react';

import { render } from '../../../test/utils/mockProvider';
import Input from './Input';

describe('Input', () => {
    it('renders correctly', () => {
        const { container } = render(<Input />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('applies disabled styles when disabled prop is true', () => {
        const { getByRole } = render(<Input disabled />);
        const inputElement = getByRole('textbox');
        expect(inputElement).toHaveAttribute('disabled');
    });
});
