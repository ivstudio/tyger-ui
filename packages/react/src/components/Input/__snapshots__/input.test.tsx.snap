// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<fieldset
  class="sc-aXZVg jsAQMJ sc-iGgWBj dIjyVK"
>
  <input
    aria-required="false"
    class="sc-eqUAAy jvnwEb"
  />
</fieldset>
`;
